### PR TITLE
Sks 74 hint support backend fix

### PR DESCRIPTION
**Pull request by megidev**

### CODE REVIEW

File: backend/flaskr/utils.py [hunk 1]
```diff
@@ -107,18 +110,25 @@ def get_scores_from_database(count=50, countries=[], gamemode="normal"):
     Returns an array of objects representing highscores in the database
     """
     # Construct SQL query
-
```
Comment (Ketsuppimakkara) [inline backend/flaskr/utils.py:110]: This change would undo the SQL injection fix. Maybe you forgot to sync the most recent changes before making edits?

File: backend/flaskr/utils.py [hunk 2]
```diff
@@ -227,10 +243,9 @@ def process_currency_data(rates):
 
 def get_exchange_rates_from_api():
     """
-    Function for getting exchange rates from Forex API.
```
Comment (Ketsuppimakkara) [inline backend/flaskr/utils.py:230]: These changes look like they are from SKS 83, which hasn't been merged yet due to some tests failing

File: backend/flaskr/utils.py [hunk 3]
```diff
@@ -286,25 +312,56 @@ def get_companies_from_database(
     Returns:
             Array of dictionaries of company data
     """
-    #Construct SQL Query
```
Comment (Ketsuppimakkara) [inline backend/flaskr/utils.py:289]: This change would undo the SQL injection fix. Maybe you forgot to sync the most recent changes before making edits?

File: backend/flaskr/utils.py [hunk 4]
```diff
@@ -416,12 +473,13 @@ def convert_marketcaps_currencies(companies, game_currency):
                     converted_market_cap = round(
                         company.get("market_cap") / (exchange_rate.get("ratio"))
                     )
```
Comment (Ketsuppimakkara) [inline backend/flaskr/utils.py:475]: I think these are part of SKS-83 too

**Review by Ketsuppimakkara — COMMENTED** [review]
These changes would get the hint data from the Yahoo Finance API, but it wouldn't update them to the database.  The hint information should be added to the database in the upsert_stock_data() function, specifically at "INSERT INTO Company (ticker, name, difficulty, market_cap, currency, date, sector, website, full_time_employees, revenue_growth, earnings_growth)"

**Comment by megidev** [comment]
Wrong branch. The updated changes have been added to a new branch.

**Pull request closed without merging**